api/app/Http/Controllers/UserController.php

```diff
@@ -1,8 +1,103 @@
-<?php
+<?php namespace App\Http\Controllers;
 
-namespace App\Http\Controllers;
+use App;
+use App\Models\User;
+use Tymon\JWTAuth\JWTAuth;
+use Illuminate\Http\Request;
+use App\Models\UserCredential;
+use Illuminate\Support\MessageBag;
+use App\Exceptions\ValidationException;
+use App\Extensions\Lock\Manager as Lock;
+use App\Repositories\UserRepository as Repository;
+use App\Http\Validators\UserValidator as Validator;
+use Spira\Responder\Contract\ApiResponderInterface as Responder;
 
-class UserController extends BaseController
+class UserController extends ApiController
 {
-    public static $model = 'App\Models\User';
+    /**
+     * Permission Lock Manager.
+     *
+     * @var Manager
+     */
+    protected $lock;
+
+    /**
+     * JWT Auth.
+     *
+     * @var Tymon\JWTAuth\JWTAuth
+     */
+    protected $jwtAuth;
+
+    /**
+     * Assign dependencies.
+     *
+     * @param  Repository  $repository
+     * @param  Lock        $lock
+     * @param  JWTAuth     $jwtAuth
+     * @param  Request     $request
+     * @param  Responder   $responder
+     * @return void
+     */
+    public function __construct(Repository $repository, Lock $lock, JWTAuth $jwtAuth, Request $request, Responder $responder)
+    {
+        $this->repository = $repository;
+        $this->lock = $lock;
+        $this->jwtAuth = $jwtAuth;
+        $this->responder = $responder;
+
+        $this->permissions($request);
+    }
+
+    /**
+     * Set permissions to be used in the controller.
+     *
+     * @param  Request  $request
+     * @return void
+     */
+    public function permissions(Request $request)
+    {
+        $this->lock->setRole(User::$userTypes);
```

Comment:
@xiphiaz Yes, it's quite a chunk of code to add to each controller for permissions currently. So definitely a lot of room for streamlining and finding a more efficient approach.